gnirehtet.cmd

```diff
@@ -0,0 +1,149 @@
+    ECHO      specified.
+    ECHO.
+    ECHO  %~nx0 uninstall [serial]
+    ECHO      Uninstall the client from the Android device and exit.
+    ECHO      If several devices are connected via adb, then serial must be
+    ECHO      specified.
+    ECHO.         
+    ECHO  %~nx0 reinstall [serial]
+    ECHO      Uninstall then install.
+    ECHO.
+    ECHO  %~nx0 rt [serial] [DNS[,DNS2,...]]
+    ECHO      Enable reverse tethering for exactly one device:
+    ECHO        - install the client if necessary;"
+    ECHO        - start the client;
+    ECHO        - start the relay server;
+    ECHO        - You'll have to stop the client and the server manually
+    ECHO.         
+    ECHO  %~nx0 start [serial] [DNS[,DNS2,...]]
+    ECHO      Start a client on the Android device and exit.
+    ECHO      If several devices are connected via adb, then serial must be
+    ECHO      specified.
+    ECHO      If DNS is given, then make the Android device use the specified
+    ECHO      DNS server(s). Otherwise, use 8.8.8.8 (Google public DNS).
+    ECHO      If the client is already started, then do nothing, and ignore
+    ECHO      DNS servers parameter.
+    ECHO      To use the host 'localhost' as DNS, use 10.0.2.2.
+    ECHO.
+    ECHO  %~nx0 stop [serial]
+    ECHO      Stop the client on the Android device and exit.
+    ECHO      If several devices are connected via adb, then serial must be
+    ECHO      specified.
+    ECHO.     
+    ECHO  %~nx0 relay
+    ECHO      Start the relay server in the current terminal.
+    ECHO.     
+    ECHO  %~nx0 killserver
+    ECHO      Kills the relay server which opened from this utility.
+    ECHO.   
+    ECHO  %~nx0 kill [serial]
+    ECHO      Kills the relay server and the Android client.
+    ECHO      If several devices are connected via adb, then serial must be
+    ECHO      specified.
+    ENDLOCAL
+    EXIT /B
+
+
+:locate_file
+    SETLOCAL EnableDelayedExpansion
+    FOR /f " tokens=1*" %%a IN ("%*") DO (
+        SET VAR_NAME=%%a
+        SET ALL_BUT_FIRST=%%b
+    )
+    FOR %%G IN (%ALL_BUT_FIRST%) DO (
+    IF EXIST %%G (
+        SET FNAME=%%G
+        )
+    )
+    IF "!FNAME!"=="" (
+        SET FNAME=%2 & ECHO cannot find file, using default name %2
+        )
+    endlocal&set %VAR_NAME%=%FNAME%
+    EXIT /B
+
+:do_install
+    ECHO 'Installing gnirehtet...'
+    @ECHO ON
+    CALL %ADB% %~1 install %APK%
+    @ECHO OFF
+    EXIT /B
+
+:do_uninstall 
+    ECHO 'Uninstall gnirehtet...'
+    @ECHO ON
+    CALL %ADB% %~1 uninstall com.genymobile.gnirehtet
+    @ECHO OFF
+    EXIT /B
+
+:do_reinstall
+    SETLOCAL
+    call :do_uninstall %1
+    call :do_install %1
+    ENDLOCAL
+    EXIT /B
+
+:do_stop
+    ECHO 'Stopping gnirehtet...'
+    CALL %ADB% %~1 shell am startservice -a com.genymobile.gnirehtet.STOP
+    EXIT /B
+
+:do_relay
+    ECHO 'Starting relay server...'
+    @ECHO ON
+    START /I "gnirehtet_relay_server" %JAVA% -jar %RELAY%
```

Comment:
There's no way of killing child process (java) via batch. I could run it the same terminal window but then I couldn't kill it easily from another process. Here I create another terminal window with unique name so I can kill it easily next time, I can hide it from the user but then he couldn't see the logs.

So my options are:
1. Start java another window. able to kill it easily, my script keeps running. User can see logs.
2. Start java another hidden window. able to kill it easily, my script keeps running. User cannot see log.
3. Start java in my own windows. cannot kill it easily from another process. my script is freezes.

What do you think?